manager/assets/modext/widgets/fc/modx.panel.fcset.js

```diff
@@ -240,22 +240,26 @@ MODx.grid.FCSetFields = function(config) {
         ,fields: ['id','action','name','tab','tab_rank','other','rank','visible','label','default_value']
         ,autoHeight: true
         ,grouping: true
-        ,groupBy: 'tab'
+        ,groupBy: 'tab_rank'
         ,plugins: [this.vcb]
         ,stateful: false
         ,remoteSort: false
         ,sortBy: 'rank'
         ,sortDir: 'ASC'
         ,hideGroupedColumn: true
-        ,groupTextTpl: '{group} ({[values.rs.length]} {[values.rs.length > 1 ? "'+_('fields')+'" : "'+_('field')+'"]})'
+        ,groupTextTpl: '{[values.rs.length]} {[values.rs.length > 1 ? "'+_('fields')+'" : "'+_('field')+'"]}'
```

Comment:
To me it doesn't make sense to remove the group name here. It now says "X fields" but it doesn't give any context about what it's referring to.

Reply:
Good point, but I couldn't display the old `tab` group there (not sure if that's possible), because fields are now grouped by `tab_rank`, not `tab`. Those if you leave `{group}` there, then `tab_rank` will be displayed.

Reply:
Why did you change the groupBy then? If you want to sort primary by the tab rank, you have to change the sorting in the processor.

Reply:
@Jako I couldn't do it, because here the question is not sorting fields, but sorting groups in extJS.
Those in whatever sort order the data arrives, the grouping sets its sorting by the group name (or I didn’t figure it out).

Reply:
If you sort in the processor before https://github.com/modxcms/revolution/blob/3.x/core/src/Revolution/Processors/Model/GetListProcessor.php#L153 in         prepareQueryBeforeCount or prepareQueryAfterCount, you can sort the result by the tab rank.

Reply:
@Jako I will not argue with you, because my knowledge is not enough, but in my opinion, you are confusing sorting and sorting groups in extJS.
Send a commit if it's easier, my version also works (and tab can be set in the group header), I'll redo my version a little later.

Reply:
Sorry, I thought it was a processor based grid. The query is created here https://github.com/modxcms/revolution/blob/3.x/manager/controllers/default/security/forms/set/update.class.php#L63-L71

I am not absolutely sure, but if you want to have the groups sorted right, you have to sort them before the lines.

Reply:
According to the code there, it is sorted by Tab.rank first, so it is the LocalGrid that sorts it wrong. Maybe it is possible to have multiple sorts in sortBy. Is it possible to remove the sorting there, because the data is already sorted right.

Reply:
As I understand it, in order to sort groups in extJS, you need to create a function that would override the default sorting, see https://stackoverflow.com/questions/28690480/extjs-5-grid-group-sorting
But in my opinion it's not worth it, my version with grouping by `tab_rank` works as it should.

Reply:
@Jako But, in general, it would be useful to know how to sort groups, it would be useful for other sections. Do you have any ideas?

Reply:
Can this discussion be resolved?

Reply:
@opengeek Done